### PR TITLE
Wip/hughsie/yara

### PR DESCRIPTION
**Pull request by hughsie**

### CODE REVIEW

File: plugins/chipsec/__init__.py [hunk 1]
```diff
@@ -147,13 +147,20 @@ def _run_chipsec_on_md(self, test, md):
         if not shards:
             blob = self._find_dell_pfs(md.blob)
             if blob:
+                shard_zlib = ComponentShard(plugin_id=self.id)
+                shard_zlib.set_blob(blob)
                 if blob.startswith(b'PFS.HDR.'):
+                    shard_zlib.ensure_info('b26f8b5c-2209-5a60-b301-39f837883a14',
```
Comment (superm1) [inline plugins/chipsec/__init__.py]: I guess this GUID is just something you came up with to assign?

File: lvfs/views_component.py [hunk 2]
```diff
@@ -694,3 +694,25 @@ def component_checksum_add(component_id):
...
+def component_download(component_id):
+
+    # find shard
+    # get firmware component
+    md = db.session.query(Component).\
+            filter(Component.component_id == component_id).first()
+    if not md:
+        flash('No component matched!', 'danger')
+        return redirect(url_for('.firmware'))
+    if not md.fw.check_acl('@view'):
```
Comment (superm1) [inline lvfs/views_component.py]: I would think you just want to leave this to folks matching researcher right?

File: plugins/blocklist/rules/bios-guard.yar [hunk 3]
```diff
@@ -0,0 +1,17 @@
+rule IntelBiosGuard
+{
+    meta:
+        license = "GPL-2.0+"
+        description = "Uses Intel BIOS Guard"
```
Comment (superm1) [inline plugins/blocklist/rules/bios-guard.yar]: I don't think you can make a claim for uses as much as a claim for "Contains"

File: plugins/blocklist/rules/bios-guard.yar [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+        long_description = "Hardware-assisted authentication and protection against BIOS recovery attacks"
+        fail = false
+        claim = "success-bios-guard"
+
+    strings:
+        $signature = "_AMIPFAT"
+        $updater = "AmiFlashUpd" wide
+        $header = "__BGKH__"
```
Comment (superm1) [inline plugins/blocklist/rules/bios-guard.yar]: Is this really enough to indicate that it's using Intel BIOS guard?  I would think it's just indicating that it has the Ami Flash updater executable for it's flasher.

File: plugins/blocklist/rules/boot-guard.yar [hunk 5]
```diff
@@ -0,0 +1,16 @@
+rule IntelBootGuard
+{
+    meta:
+        license = "GPL-2.0+"
+        description = "Uses Intel Boot Guard"
```
Comment (superm1) [inline plugins/blocklist/rules/boot-guard.yar]: Again, I think "Contains" is more accurate in this instance.

File: plugins/blocklist/rules/bios-guard.yar (hunk 4 above)
Comment (hughsie) [reply plugins/blocklist/rules/bios-guard.yar]: It's `$header or ($signature and not $updater)` -- i.e. we exclude the updater.

File: plugins/blocklist/rules/bios-guard.yar (hunk 3 above)
Comment (hughsie) [reply plugins/blocklist/rules/bios-guard.yar]: True.

File: lvfs/views_component.py (hunk 2 above)
Comment (hughsie) [reply lvfs/views_component.py]: I wondered if it was useful for anything else -- other @view users can just download the .cab file and extract it already.

File: plugins/chipsec/__init__.py (hunk 1 above)
Comment (hughsie) [reply plugins/chipsec/__init__.py]: It's `generate-guid PFS` -- if you have something more official please let me know :)

File: plugins/blocklist/rules/bios-guard.yar (hunk 4 above)
Comment (superm1) [reply plugins/blocklist/rules/bios-guard.yar]: Oh I see.  OK.

File: lvfs/views_component.py (hunk 2 above)
Comment (superm1) [reply lvfs/views_component.py]: Well can always expand it to more people later if there is a need though too.  I think by keeping it researcher only you'll be more likely to looped into use cases that people want it.

File: plugins/chipsec/__init__.py (hunk 1 above)
Comment (superm1) [reply plugins/chipsec/__init__.py]: Ah OK.

I have no idea where the name `PFS` even came from.  Whoever first reverse engineered one of the implementations maybe?

Comment (hughsie) [reply plugins/chipsec/__init__.py]: It's the first three bytes of the decompressed Zlib section.

File: lvfs/views_component.py (hunk 2 above)
Comment (hughsie) [reply lvfs/views_component.py]: Fair point, will fix.

**Pull request closed without merging**